src/Compiler/CompilerOptions.cpp

```diff
@@ -303,32 +452,62 @@ llvm::cl::opt<std::string> modelTag("tag",
         "`--tag` will disable tag completely, meaning no tag is appended to "
         "the symbols."),
     llvm::cl::value_desc("a string that matches regex ([0-9a-z_.-]+)"),
-    llvm::cl::init(""), llvm::cl::cat(OnnxMlirOptions));
+    llvm::cl::location(modelTag), llvm::cl::init(""),
+    llvm::cl::cat(OnnxMlirOptions));
 
-llvm::cl::opt<bool> enableConvOptPass("enable-conv-opt-pass",
+static llvm::cl::opt<bool, true> enableConvOptPassOpt("enable-conv-opt-pass",
     llvm::cl::desc("Enable the ConvOptPass. Default is true."),
-    llvm::cl::init(true), llvm::cl::cat(OnnxMlirOptions));
+    llvm::cl::location(enableConvOptPass), llvm::cl::init(true),
+    llvm::cl::cat(OnnxMlirOptions));
 
-llvm::cl::list<std::string> extraLibPaths("L",
+static llvm::cl::list<std::string, std::vector<std::string>> extraLibPathsOpt(
+    "L",
     llvm::cl::desc("Specify extra directories for libraries when compiling"
                    "an onnx model. Will be add used as -L in the linkage step."
                    "Each directory can be specified with one extra-lib-dirs"),
-    llvm::cl::Prefix, llvm::cl::cat(OnnxMlirOptions));
+    llvm::cl::location(extraLibPaths), llvm::cl::Prefix,
+    llvm::cl::cat(OnnxMlirOptions));
 
-llvm::cl::list<std::string> extraLibs("l",
+static llvm::cl::list<std::string, std::vector<std::string>> extraLibsOpt("l",
     llvm::cl::desc("Specify extra libraries when compiling an onnx model."
                    "Will be add used as -l in the linkage step."
                    "Each lib can be specified with one extra-libs"),
-    llvm::cl::Prefix, llvm::cl::cat(OnnxMlirOptions));
+    llvm::cl::location(extraLibs), llvm::cl::Prefix,
+    llvm::cl::cat(OnnxMlirOptions));
 
-llvm::cl::opt<ProfileIRs> profileIR("profile-ir",
+static llvm::cl::opt<ProfileIRs, true> profileIROpt("profile-ir",
     llvm::cl::desc("Profile operations in an IR"),
+    llvm::cl::location(profileIR),
     llvm::cl::values(clEnumVal(None, "No profiling. Default value."),
         clEnumVal(
             Onnx, "Profile operations in ONNXIR generated by --EmitONNXIR.")
             APPLY_TO_ACCELERATORS(ACCEL_PROFILEIR_CL_ENUM)),
     llvm::cl::init(ProfileIRs::None), llvm::cl::cat(OnnxMlirOptions));
 
+// Options for onnx-mlir-opt only
```

Comment:
High level question: so some of these options were only declared in a cpp file for `onnx-mlir` and others only for `onnx-mlir-opt`. Are we now defining all of these for both? Does it means that when we type `--help` for either, we get the union of all options, including some that don't apply for that program?
If so, that would be unfortunate.

Reply:
`--help` will only show common and command specific options. However, as indicated above, you can still use an option of the other command which is silently ignored.